### PR TITLE
Bsongis/horus xjt external module

### PR DESCRIPTION
**Pull request by bsongis**
Here is a refactoring for the pulses driver for Horus / Taranis
- Now external module should be OK on Horus
- Taranis RAM saving (2x400bytes)

The DMA Stream used are now for TIMx_UP

### CODE REVIEW

File: radio/src/pulses/dsm2_arm.cpp [hunk 1]
```diff
@@ -140,7 +137,7 @@ void setupPulsesDSM2(unsigned int port)
       break;
   }
 
-#if !defined(PPM_PIN_HW_SERIAL)
+#if !defined(PPM_PIN_SERIAL)
   if (moduleFlag[port] == MODULE_BIND)
```
Comment (schwabe) [inline radio/src/pulses/dsm2_arm.cpp]: Althought not really part of this commit, this ifdef looks strange. Binding not supported on Horus?

Comment (bsongis) [inline radio/src/pulses/dsm2_arm.cpp]: Yes it is, ```PPM_PIN_SERIAL``` is not defined on Horus

Comment (bsongis) [inline radio/src/pulses/dsm2_arm.cpp]: But that code is indeed strange, let-me rewrite it a little bit

File: radio/src/pulses/pulses_arm.h [hunk 2]
```diff
@@ -46,25 +52,44 @@ PACK(struct PxxPulsesData {
...
+#if defined(PPM_PIN_UART)
+PACK(struct PxxUartPulsesData {
+  uint8_t  pulses[64];
+  uint8_t  * ptr;
+  uint16_t pcmCrc;
+});
+#endif
+
+/* PXX uses 20 bytes (as of Rev 1.1 document) with 8 changes per byte + stop bit ~= 162 max pulses */
+#if defined(PPM_PIN_TIMER)
```
Comment (schwabe) [inline radio/src/pulses/pulses_arm.h]: That define is only used here, maybe use a #if !defined(PPM_PIN_UART)?

Comment (bsongis) [inline radio/src/pulses/pulses_arm.h]: Horus has PPM_PIN_UART and PPM_PIN_TIMER!

File: radio/src/pulses/pulses_arm.h [hunk 3]
```diff
@@ -69,28 +69,32 @@ PACK(struct PxxUartPulsesData {
...
-  pulse_duration_t pulses[300];
-#else
-  pulse_duration_t pulses[200];
-#endif
+  pulse_duration_t pulses[NUM_TIMER_PULSES];
   pulse_duration_t * ptr;
   uint16_t rest;
   uint8_t index;
 });
+#undef NUM_TIMER_PULSES
```
Comment (bsongis) [inline radio/src/pulses/pulses_arm.h]: What is it?

File: radio/src/pulses/pulses_arm.h [hunk 4]
```diff
@@ -69,28 +69,32 @@ PACK(struct PxxUartPulsesData {
...
+/* Multimodule reuses some of the DSM2 function and structs since the protocols are similar enough */
+#if defined(MULTIMODULE)
+#define NUM_TIMER_PULSES 300
+#else
+#define NUM_TIMER_PULSES 200
+#endif
+
 PACK(struct PxxTimerPulsesData {
-  pulse_duration_t pulses[200];
+  pulse_duration_t pulses[NUM_TIMER_PULSES];
```
Comment (bsongis) [inline radio/src/pulses/pulses_arm.h]: Why? Does multimodule use the pxx part of the union, I thought it was only dsm2

Comment (schwabe) [inline radio/src/pulses/pulses_arm.h]: For some reason that I don't understand the pulses for PXX only work if the DSM2 part and the pxx part have the same location of the ptr member

**Comment by bsongis** [comment]
Explanation from Mike for the +/- 2uS:

If I've read it correctly for pxx, you are sending pulses 8uS wide **(with some skew to allow for the transistor switching delay)**

**Comment by bsongis** [comment]
This is not ready to merge. I was about to fly with this version when the plane stopped to answer. It became again OK 3-4s later, then again KO. 

I had to flash the 'next' branch to fly ...

So at least the internal PXX pulses are broken

File: radio/src/pulses/pxx_arm.cpp [hunk 5]
```diff
@@ -148,16 +148,17 @@ void pxxPutPcmTail(uint8_t port)
 void pxxPutPcmBit(uint8_t port, uint8_t bit)
 {
   if (bit) {
-    modulePulsesData[port].pxx.pcmOnesCount += 1;
     pxxPutPcmPart(port, 1);
+    if (++modulePulsesData[port].pxx.pcmOnesCount == 5) {
+      modulePulsesData[port].pxx.pcmOnesCount = 0;
```
Comment (bsongis) [inline radio/src/pulses/pxx_arm.cpp]: Here!
